pkg/auth/oauth2/auth_openshift.go

```diff
@@ -229,12 +232,85 @@ func (o *openShiftAuth) Authenticate(_ http.ResponseWriter, r *http.Request) (*a
 	if err != nil {
 		return nil, err
 	}
+
 	if cookie.Value == "" {
 		return nil, fmt.Errorf("unauthenticated, no value for cookie %s", sessions.OpenshiftAccessTokenCookieName)
 	}
 
+	tokenReviewURL, err := url.Parse(o.issuerURL)
+	if err != nil {
+		klog.Errorf("failed to parse the URL to kube-apiserver: %v", err)
+		return nil, err
+	}
+	tokenReviewURL.Path = "/apis/authentication.k8s.io/v1/tokenreviews"
+	tokenReviewURL.RawPath = "/apis/authentication.k8s.io/v1/tokenreviews"
+
+	tokenReview := &authv1.TokenReview{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "authentication.k8s.io/v1",
+			Kind:       "TokenReview",
+		},
+		Spec: authv1.TokenReviewSpec{
+			Token: cookie.Value,
+		},
+	}
+
+	tokenReviewJSON, err := json.Marshal(tokenReview)
+	if err != nil {
+		klog.Errorf("failed to marshal TokenReview to JSON: %v", err)
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, tokenReviewURL.String(), bytes.NewBuffer(tokenReviewJSON))
+	if err != nil {
+		klog.Errorf("failed to create TokenReview request: %v", err)
+		return nil, err
+	}
+
+	// Set headers for the HTTP request
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", r.Header.Get("Authorization"))
+
+	res, err := o.k8sClient.Do(req)
+	if err != nil {
+		klog.Errorf("failed to execute TokenReview request: %v", err)
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusCreated {
```

Comment:
is it really only 201 ?

Reply:
That's what my testing showed. When successful, we get a 201 response